Community.ManagementPackCatalog/CheckForManagementPackUpdates.ps1

```diff
@@ -0,0 +1,80 @@
+﻿# CheckForManagementPackUpdates.ps1 is part of the Community Management Pack Catalog and licensed GPL v3; See https://github.com/mpcatalog/scom-community-catalog-ui/blob/master/LICENSE.
+
+# Import the OpsMgr Module and connect to localhost this will target
+# the RMS Emulator so we know it will be a Management Server
+
+Import-Module OperationsManager
+New-SCOMManagementGroupConnection localhost
+
+# Create a API object which will be used to report the status back to the monitor
+$api = New-Object -Com MOM.ScriptAPI
+$bag = $api.CreatePropertyBag()
+
+# Use the redirect to pull out base repository location
+$redirectBaseUrl = "http://www.mpcatalog.net/CatalogRepo";
+$gitHubRepoBase = (Invoke-WebRequest -Uri $redirectBaseUrl -MaximumRedirection 0 -ErrorAction SilentlyContinue -Headers @{"Referer"="Http://Ps.UpdateMonitor"}).Headers.Location;
+
+# Test to see if this is a secondary redirect, we will follow a maximum of two
+$redirectTestResult = (Invoke-WebRequest -Uri $gitHubRepoBase -MaximumRedirection 0 -ErrorAction SilentlyContinue -Headers @{"Referer"="Http://Ps.UpdateMonitor"})
+if ($redirectTestResult.StatusCode -eq 301 -or $redirectTestResult.StatusCode -eq 302)
+	{
+		# If this is another redirect, we'll update the repo address.  If not, do nothing.
+		$gitHubRepoBase = $redirectTestResult.Headers.Location
+	}
+
+$managementPackIndexUrl = $gitHubRepoBase + "Index.json";
+
+# Get list of management packs to work with
+$mpsFromGitHub = Invoke-RestMethod -Uri $managementPackIndexUrl -Method Get;
+
+[bool]$OutdatedPacks = $false;
+$StatusMessage = [string]::Empty
+
+# Check each management pack in the list for status
+foreach($mp in $mpsFromGitHub)
+    {
+		# Get the details from GitHub for this specific pack.
+        $detailsJsonFileLocation = $gitHubRepoBase + $mp.ManagementPackSystemName + "/details.json";
+        $PackDetails = Invoke-RestMethod -Uri $detailsJsonFileLocation -Method Get | Select ManagementPackSystemName,@{n="Version"; e={[version]::Parse($_.Version)}},ManagementPackDisplayName
+        
+		# Pull the management pack via SDK, with $null as the fall back
+		$InstalledMp = $null
+        $InstalledMp = Get-SCOMManagementPack -Name $PackDetails.ManagementPackSystemName
+
+		# If the pack is installed, compare the version against GitHub.  For packs with updates available, add to the OutdatedPacks counter.
+		If($InstalledMp -ne $null)
+            {
+                If($PackDetails.Version -gt $InstalledMp.Version)
```

Comment:
Will this do a proper version comparison?

Reply:
The PowerShell version comparison works great, I tried multiple variations using code similar to below and always received accurate results.

[version]$v1 = [version]::Parse("1.0.1.19")
[version]$v2 = [version]::Parse("1.1.0.3")
$v1 -le $v2

Reply:
Ah hadn't noticed you were just hooking into System.Version.Parse()